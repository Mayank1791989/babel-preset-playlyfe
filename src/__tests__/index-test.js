/* @flow */
import preset from '../index';

describe('preset-options', () => {
  it('works without options', () => {
    expect(() => {
      preset();
    }).not.toThrow();
  });

  it('dont throw when options are correct', () => {
    expect(() => {
      preset(null, { react: true });
    }).not.toThrow();
  });

  it('throws when unknown options passed', () => {
    expect(() => {
      // $FlowDisable - testing this error
      preset(null, { xyz: true });
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws when wrong value passed for option', () => {
    expect(() => {
      // $FlowDisable - testing this error
      preset(null, { react: 'xyz' });
    }).toThrowErrorMatchingSnapshot();
  });
});
