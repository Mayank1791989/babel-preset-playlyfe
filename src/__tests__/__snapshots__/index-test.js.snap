// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-options throws when unknown options passed 1`] = `
"unknown option passed 'xyz'
Available opts
{
  flow: boolean;
  asyncAwait: boolean;
  dynamicImport: boolean;
  modules: \\"amd\\" | \\"umd\\" | \\"systemjs\\" | \\"commonjs\\" | false;
  targets: Array<any>;
  decorators: boolean;
  useBuiltIns: boolean;
  debug: boolean;
  reactIntl: boolean;
  react: boolean;
  babelRuntime: false;
}"
`;

exports[`preset-options throws when wrong value passed for option 1`] = `
"opts.react must be true or false

Expected: boolean

Actual: string
"
`;
